keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+	Kind string `json:"kind"`
+
+	// Namespace is the namespace of the referent.
+	Namespace string `json:"namespace"`
+}
+
+// ReferenceGrantTo describes what Kinds are allowed as targets of the
+// references.
+type ReferenceGrantTo struct {
+	// Group is the group of the referent.
+	// When empty, the Kubernetes core API group is inferred.
+	Group string `json:"group"`
+
+	// Kind is the kind of the referent.
+	Kind string `json:"kind"`
+
+	// Name is the name of the referent. When unspecified, this policy
+	// refers to all resources of the specified Group and Kind in the local
+	// namespace.
+	//
+	// +optional
+	Name *string `json:"name,omitempty"`
+}
+```
+
+### Outstanding questions and clarifications
+
+This section lists some of the outstanding questions people have had about
+ReferenceGrant, and other items that we'll be clarifying in the godoc and other
+documentation as part of the transition to the new API group, keeping in mind
+that we will not be adding any new features, fields, or behavior, just clarifying
+what's done already.
+
+Also note that we don't consider any of these blockers for the general _idea_ of
+moving ReferenceGrant to the new API group, just notes to save discussion time.
+
+* Clarify that an implementation is required to reconcile ReferenceGrant for
+  specific `To` Kinds.
+* Corollary for future work, define how controllers interact. Is it a problem if
+  multiple controllers consume the same ReferenceGrant?
+* Status design is still pending, but it's currently expected that controllers
+  will indicate status on the _referring_ resources, not on ReferenceGrant itself.
+* Clarify that the expected operating model for implementations expects them to
+  have broad, read access to both ReferenceGrant and the specific `To` Kinds they
+  support, and then self-limit to only _use_ the relevant ones.
+* Decide whether to formally add `*` as a special value for Namespace, to mean
+  "all namespaces".
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+This is a net new resource to Kubernetes so it will not require any changes or
+additions to existing tests.
+
+##### Unit tests
+
+Unit tests will be used to cover:
+
+1. ReferenceGrant validation
+2. ReferenceGrant implementation library
+
+More details will be added as the details of the implementation library are
+clarified.
+
+##### Integration tests
+
+Because we are not bundling an implementation with this API, no integration
+tests are possible.
+
+##### e2e tests
+
+We will strongly encourage every API that uses ReferenceGrant to define
+conformance tests for their use of ReferenceGrant.
+
+### Graduation Criteria
+
+#### GA
+
+[x] Almost all of the fields and behavior have conformance test coverage.
+[x] Multiple conformant implementations.
```

Comment:
this doesn't seem like it can be checked quite yet :)

Reply:
Woops, thanks, it is true of the current conformance test coverage of the Gateway API resource, but given the changes this KEP will result in, it definitely is not true now, will update.